components/movementsensor/wheeledodometry/wheeledodometry_test.go

```diff
@@ -110,6 +116,51 @@ func TestNewWheeledOdometry(t *testing.T) {
 	test.That(t, ok, test.ShouldBeTrue)
 }
 
+func TestReconfigure(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+
+	deps := make(resource.Dependencies)
+	deps[base.Named(baseName)] = createFakeBase(0.1, 0.1, 0)
+	deps[motor.Named(leftMotorName)] = createFakeMotor(true)
+	deps[motor.Named(rightMotorName)] = createFakeMotor(false)
+
+	fakecfg := resource.Config{
+		Name: testSensorName,
+		ConvertedAttributes: &Config{
+			LeftMotors:        []string{leftMotorName},
+			RightMotors:       []string{rightMotorName},
+			Base:              baseName,
+			TimeIntervalMSecs: 500,
+		},
+	}
+	fakeSensor, err := newWheeledOdometry(ctx, deps, fakecfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+	od, ok := fakeSensor.(*odometry)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	newDeps := make(resource.Dependencies)
+	newDeps[base.Named(newBaseName)] = createFakeBase(0.2, 0.2, 0)
+	newDeps[motor.Named(newLeftMotorName)] = createFakeMotor(true)
+	newDeps[motor.Named(newRightMotorName)] = createFakeMotor(false)
+
+	newconf := resource.Config{
+		Name: testSensorName,
+		ConvertedAttributes: &Config{
+			LeftMotors:        []string{newLeftMotorName},
+			RightMotors:       []string{newRightMotorName},
+			Base:              newBaseName,
+			TimeIntervalMSecs: 300,
+		},
+	}
+
+	err = fakeSensor.Reconfigure(ctx, newDeps, newconf)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, od.timeIntervalMSecs, test.ShouldEqual, 300)
+	test.That(t, od.baseWidth, test.ShouldEqual, 0.2)
+	test.That(t, od.wheelCircumference, test.ShouldEqual, 0.2)
```

Comment:
optional but might want to add another test case where not everything in the config changes and testing that the attributes that are supposed to stay the same don't change. but if you've also done manual testing for this then prob fine!

Reply:
good point, will do!